katz/src/main/kotlin/katz/data/Try.kt

```diff
@@ -74,26 +78,35 @@ sealed class Try<out A> {
      */
     fun <B> fold(fa: (Throwable) -> B, fb: (A) -> B): B = when (this) {
         is Failure -> fa(exception)
-        is Success -> try { fb(value) } catch (e: Throwable) { fa(e) }
+        is Success -> try {
+            fb(value)
+        } catch (e: Throwable) {
+            fa(e)
+        }
     }
 
     /**
      * The `Failure` type represents a computation that result in an exception.
      */
-    class Failure<out A>(val exception: Throwable) : Try<A>() {
-        override val isFailure: Boolean = false
-        override val isSuccess: Boolean = true
+    data class Failure<out A>(val exception: Throwable) : Try<A>() {
+        override val isFailure: Boolean = true
+        override val isSuccess: Boolean = false
```

Comment:
These booleans look like a possible scape hatch we don't really need. They could either be retrieved by a fold, or simply using when.